examples/prometheus_exporter/main.cc

```diff
@@ -17,124 +17,123 @@ namespace prom_exp   = opentelemetry::exporter::prometheus;
+  }
 
-void histogram_example(prom_exp::PrometheusExporter &exporter)
-{
-  std::vector<double> boundaries{10, 20, 30, 40, 50};
-  auto aggregator = std::shared_ptr<metric_sdk::Aggregator<int>>(
-      new metric_sdk::HistogramAggregator<int>(metric_api::InstrumentKind::Counter, boundaries));
-
-  // collection of records to export
-  std::vector<metric_sdk::Record> collection;
-  auto record = get_record("histogram", 1, "{label-1:v1,label2:v2,}", aggregator);
-  do_example(aggregator, collection, record, exporter);
-}
+  void histogram_example(prom_exp::PrometheusExporter &exporter)
+  {
+    std::vector<double> boundaries{10, 20, 30, 40, 50};
+    auto aggregator = std::shared_ptr<metric_sdk::Aggregator<int>>(
+        new metric_sdk::HistogramAggregator<int>(metric_api::InstrumentKind::Counter, boundaries));
+
+    // collection of records to export
+    std::vector<metric_sdk::Record> collection;
+    auto record = get_record("histogram", 1, "{label-1:v1,label2:v2,}", aggregator);
+    do_example(aggregator, collection, record, exporter);
+  }
 
-void exact_example(prom_exp::PrometheusExporter &exporter)
-{
-  auto aggregator = std::shared_ptr<metric_sdk::Aggregator<int>>(
-      new metric_sdk::ExactAggregator<int>(metric_api::InstrumentKind::Counter, true));
+  void exact_example(prom_exp::PrometheusExporter &exporter)
+  {
+    auto aggregator = std::shared_ptr<metric_sdk::Aggregator<int>>(
+        new metric_sdk::ExactAggregator<int>(metric_api::InstrumentKind::Counter, true));
 
-  // collection of records to export
-  std::vector<metric_sdk::Record> collection;
-  auto record = get_record("exact", 1, "{label-1:v1,label2:v2,}", aggregator);
-  do_example(aggregator, collection, record, exporter);
-}
+    // collection of records to export
+    std::vector<metric_sdk::Record> collection;
+    auto record = get_record("exact", 1, "{label-1:v1,label2:v2,}", aggregator);
+    do_example(aggregator, collection, record, exporter);
+  }
 
-void sketch_example(prom_exp::PrometheusExporter &exporter)
-{
-  auto aggregator = std::shared_ptr<metric_sdk::Aggregator<int>>(
-      new metric_sdk::SketchAggregator<int>(metric_api::InstrumentKind::Counter, true));
+  void sketch_example(prom_exp::PrometheusExporter &exporter)
+  {
+    auto aggregator = std::shared_ptr<metric_sdk::Aggregator<int>>(
+        new metric_sdk::SketchAggregator<int>(metric_api::InstrumentKind::Counter, true));
 
-  // collection of records to export
-  std::vector<metric_sdk::Record> collection;
-  auto record = get_record("sketch", 1, "{label-1:v1,label2:v2,}", aggregator);
-  do_example(aggregator, collection, record, exporter);
+    // collection of records to export
+    std::vector<metric_sdk::Record> collection;
+    auto record = get_record("sketch", 1, "{label-1:v1,label2:v2,}", aggregator);
+    do_example(aggregator, collection, record, exporter);
+  }
 }
-
-int main()
+int main(int argc, char** argv)
 {
+  std::string agg(argv[1]);
```

Comment:
Please handle argc < 2 before doing this.

Reply:
Missed this earlier, but fixed now.